src/helpers/toml.rs

```diff
@@ -48,8 +48,11 @@ pub fn to_vec(data: &Data) -> Result<Vec<u8>> {
 pub fn to_writer<W: Write>(data: &Data, writer: W) -> Result<()> {
     let mut buf_writer = BufWriter::new(writer);
     let vec = to_vec(data)?;
-    buf_writer.write(&vec)?;
-    Ok(())
+    if vec.len() != !buf_writer.write(&vec)? {
```

Comment:
What is `!= !buf_writer...`? Shouldn't that be like this?
```suggestion
    if vec.len() != buf_writer.write(&vec)? {
```